cmd/test.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/tester"
+	"github.com/spf13/cobra"
+)
+
+var testParams = struct {
+	verbose  bool
+	errLimit int
+}{}
+
+var testCommand = &cobra.Command{
+	Use:   "test",
+	Short: "Execute Rego test cases",
+	Long: `Execute Rego test cases.
+
+The 'test' command takes a file or directory path as input and executes all
+test cases discovered in matching files. Test cases are rules whose names have the prefix "test_".
+
+Example policy (example/authz.rego):
+
+	package authz
+
+	allow {
+		input.path = ["users"]
+		input.method = "POST"
+	}
+
+	allow {
+		input.path = ["users", profile_id]
+		input.method = "GET"
+		profile_id = input.user_id
+	}
+
+Example test (example/authz_test.rego):
+
+	package authz
+
+	test_post_allowed {
+		allow with input as {"path": ["users"], "method": "POST"}
+	}
+
+	test_get_denied {
+		not allow with input as {"path": ["users"], "method": "GET"}
+	}
+
+	test_get_user_allowed {
+		allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "bob"}
+	}
+
+	test_get_another_user_denied {
+		not allow with input as {"path": ["users", "bob"], "method": "GET", "user_id": "alice"}
+	}
+
+Example test run:
+
+	$ opa test ./example/
+`,
+	Run: func(cmd *cobra.Command, args []string) {
+		os.Exit(opaTest(args))
+	},
+}
+
+func opaTest(args []string) int {
+	compiler := ast.NewCompiler().SetErrorLimit(testParams.errLimit)
+	results, err := tester.NewRunner().SetCompiler(compiler).Run(args...)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
+		return 1
+	} else if len(results) == 0 {
+		fmt.Fprintln(os.Stderr, "warning: no tests found")
+		return 0
+	}
+	reporter := tester.PrettyReporter{
+		Verbose: testParams.verbose,
+		Output:  os.Stdout,
+	}
+	if err := reporter.Report(results); err != nil {
```

Comment:
Would be valuable to print results as they finish, as a way of marking progress and knowing where the tests get stuck or take a long time.  Could also be nice to have timing-data returned.